lib/datapacksbuilder.js

```diff
@@ -326,7 +326,20 @@ DataPacksBuilder.prototype.setFileData = async function(filePath, filePath, enco
     if (!this.allFileDataMap) {
         this.allFileDataMap = {};
     }
+
     this.allFileDataMap[path.normalize(filePath).toLowerCase()] = data;
+
+    try {
+        var dataPack = JSON.parse(data);
+
+        if (dataPack && dataPack.VlocityRecordSourceKey) {
+            if (!this.recordSourceKeyToFilePath) {
+                this.recordSourceKeyToFilePath = {};
+            }
+
+            this.recordSourceKeyToFilePath[dataPack.VlocityRecordSourceKey] = path.normalize(filePath);
```

Comment:
There is more than 1 file per dataPack.VlocityRecordSourceKey, so this will not be a correct flow.